### PR TITLE
feat: added TagBase component

### LINKED ISSUE #9465: [MOB] Add TagBase component

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
This PR introduces the TagBase component, laying the groundwork for a unified approach to various tag components within our system, including `TagColored`, `Tag`, `TagURL`, and `Name`. This update is pivotal for enhancing consistency and reusability across our UI components. Below, I'll detail the context, purpose, and justification for these changes, alongside a brief overview of the key properties introduced.

## **Context and Knowledge for Review**
To effectively review this PR, understanding the current inconsistencies and limitations in our tag components is crucial. Previously, similar functionalities and styles were replicated across different components, leading to code duplication and a lack of standardization. The design resources linked below provide a visual reference for the intended outcomes of these updates.
- Design Resources: https://www.figma.com/file/HKpPKij9V3TpsyMV1TpV7C/DS-Components?type=design&node-id=26837-1400&mode=design&t=NR9466btb5nd8KCN-11

## **Purpose of the PR**
The primary goal of this PR is to:
- Consolidate tag-related functionalities into a single base component (TagBase), promoting code reuse and consistency.
- Introduce a flexible system that accommodates the needs of various tag components through a unified interface.
This initiative is driven by the need for a more streamlined, maintainable, and scalable approach to managing tag components within our design system.

## **How the PR Satisfies the Need**
The TagBase component addresses these needs by offering:
- *Key Props:*
  - startAccessory: Enhances visual flexibility, used by TagURL, Name, and Tag.
  - endAccessory: Enhances visual flexibility, used by TagURL.
  - shape: Provides shape variations for all tags, supporting diverse design requirements.
  - severity: Introduces color-coded severity levels for tags, applicable across all tag types.
- *Extended Functionality in Other Components:*
  - The `ListItem` component has been updated to accept additional props from `ViewProps`, increasing its versatility.
  - The `TextVariant` enum now includes text variants from `design-tokens`, aligning text styles more closely with our design system.

## **Related issues**

Fixes: #9465 

## **Manual testing steps**

1. Go to storybook
2. Check Component Library Base
3. Check TagBase

### **Before**
The Before screenshot does not exist

### **After**
https://github.com/MetaMask/metamask-mobile/assets/14355083/99c129a1-2993-42e0-b930-7d35931ae5d1

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by brianacnguyen** [comment]
This component is created as a base to `Tag`, `TagURL`, `TagColored`, and `Name` components

File: app/component-library/components/List/ListItemMultiSelect/ListItemMultiSelect.tsx [hunk 1]
```diff
@@ -23,10 +23,9 @@ const ListItemMultiSelect: React.FC<ListItemMultiSelectProps> = ({
   ...props
 }) => {
   const { styles } = useStyles(styleSheet, { style, gap, isDisabled });
-  const { hitSlop, ...listItemProps } = props;
   return (
     <TouchableOpacity style={styles.base} disabled={isDisabled} {...props}>
-      <ListItem gap={gap} style={styles.listItem} {...listItemProps}>
+      <ListItem gap={gap} style={styles.listItem}>
```
Comment (brianacnguyen) [inline app/component-library/components/List/ListItemMultiSelect/ListItemMultiSelect.tsx:28]: this is to fix unintended errors with duplicated testIDs

File: app/component-library/components/List/ListItemMultiSelect/ListItemMultiSelect.types.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ import { ListItemProps } from '../../List/ListItem/ListItem.types';
  */
 export interface ListItemMultiSelectProps
   extends TouchableOpacityProps,
-    Omit<ListItemProps, 'style'> {
+    Omit<ListItemProps, 'style' | 'hitSlop'> {
```
Comment (brianacnguyen) [inline app/component-library/components/List/ListItemMultiSelect/ListItemMultiSelect.types.ts:12]: fixing lint error

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Great PR description, walkthrough video and comprehensive testing! 
- Checked code ✅ 
- Checked spelling, gramma and links work in readme ✅ 

Wasn't able to run storybook locally because of local build issues but confident in the updates because of excellent video walkthrough

File: app/component-library/base-components/TagBase/TagBase.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        type: 'select',
+      },
+    },
+    gap: {
+      control: {
+        type: 'number',
+        min: 0,
+      },
+    },
+    showSampleStartAccessory: {
```
[the remark this answers is not in the document]
Comment (brianacnguyen) [reply app/component-library/base-components/TagBase/TagBase.stories.tsx:35]: Will update

**Review by georgewrmarshall — DISMISSED** [review]
Nicely done 💯 🚀

File: app/component-library/components/Texts/Text/Text.types.ts [hunk 4]
```diff
@@ -13,10 +13,13 @@ export enum TextVariant {
   HeadingSM = 'sHeadingSM',
   BodyLGMedium = 'sBodyLGMedium',
   BodyMD = 'sBodyMD',
+  BodyMDMedium = 'sBodyMDMedium',
   BodyMDBold = 'sBodyMDBold',
   BodySM = 'sBodySM',
+  BodySMMedium = 'sBodySMMedium',
   BodySMBold = 'sBodySMBold',
   BodyXS = 'sBodyXS',
+  BodyXSMedium = 'sBodyXSMedium',
```
Comment (brianacnguyen) [inline app/component-library/components/Texts/Text/Text.types.ts:22]: `BodySMMedium` was needed as a default for the `TagBase`, so updated the `TextVariant` enum with updated variants from `design-tokens`

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**